tests/pytest/core/test_middleware_login_required.py

```diff
@@ -0,0 +1,56 @@
+from django.urls import reverse
+from django.utils.decorators import decorator_from_middleware
+
+import pytest
+
+from benefits.core import session
+from benefits.core.middleware import LoginRequired
+from benefits.core.models import EligibilityVerifier
+
+login_path = reverse("oauth:login")
+
+
+@pytest.fixture
+def decorated_view(mocked_view):
+    return decorator_from_middleware(LoginRequired)(mocked_view)
+
+
+def require_login(request):
+    verifier = EligibilityVerifier.objects.filter(auth_provider__isnull=False).first()
+    assert verifier
+    assert verifier.requires_authentication
+    session.update(request, verifier=verifier)
+
+
+@pytest.mark.django_db
+def test_login_auth_required(app_request, mocked_view, decorated_view):
+    require_login(app_request)
+    response = decorated_view(app_request)
+
+    mocked_view.assert_not_called()
```

Comment:
Is this the same `mocked_view` that was `decorated` ... because of the order of the fixture args here?

Reply:
[pytest fixtures are cached on a test-by-test basis](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached); order doesn't matter.